jscomp/ml/typecore.ml

```diff
@@ -798,10 +801,36 @@ end) = struct
     | Tconstr(p, _, _) -> p
     | _ -> assert false
 
-  let lookup_from_type env tpath lid =
+  let lookup_from_type env tpath (lid : Longident.t loc) : Name.t =
     let descrs = get_descrs (Env.find_type_descrs tpath env) in
     Env.mark_type_used env (Path.last tpath) (Env.find_type tpath env);
-    match lid.txt with
+    let is_dict = Path.same tpath Predef.path_dict in
+    if is_dict then (
+      (* [dict] Dicts are implemented as a record with a single "magic" field. This magic field is 
```

Comment:
this is getting a bit repetitive, the same information in 3 places

Reply:
I'll center it to one place and just leave small breadcrumbs in places like this.